source/ed/build_helper.h

```diff
@@ -82,6 +85,57 @@ struct SA {
     SA & operator()(const std::string & sp_name, double x = 0, double y = 0, bool wheelchair_boarding = true);
 };
 
+//     b.add_impact().name("bob").period("12d").severity(NoService).on(stop_1).on(line1);
+struct DisruptionCreator;
+
+struct Impacter {
+    Impacter(builder&, nt::disruption::Disruption&);
+    builder& b;
+    boost::shared_ptr<nt::disruption::Impact> impact;
+    const nt::disruption::Disruption& get_disruption() const {
+        return *impact->disruption;
+    }
+    Impacter& uri(const std::string& u) { impact->uri = u; return *this; }
+    Impacter& application_periods(const boost::posix_time::time_period& p) {
+        impact->application_periods.push_back(p);
+        return *this;
+    }
+    Impacter& severity(nt::disruption::Effect,
+                       std::string uri = "",
+                       const std::string& wording = "",
+                       const std::string& color = "#FFFF00",
+                       int priority = 0);
+
+    Impacter& severity(const std::string& uri); // link to existing severity
+    Impacter& informed_entities(nt::Type_e type, const std::string& uri);
+    Impacter& on(nt::Type_e type, const std::string& uri); // syntaxic sugar to informed_entities
+    Impacter& msg(const nt::disruption::Message&);
+    Impacter& msg(const std::string& msg, nt::disruption::ChannelType = nt::disruption::ChannelType::email);
+    Impacter& publish(const boost::posix_time::time_period& p) {
```

Comment:
I think publish is a bit nicer to use, but if you think it not clear I can change it